invoice.go

```diff
@@ -215,3 +215,27 @@ func (ir *InvoiceRequest) Download(ctx context.Context, invoiceID string) (*Invo
 
 	return nil, nil
 }
+
+func (ir *InvoiceRequest) RetryPayment(ctx context.Context, invoiceID string) (*Invoice, *Error) {
+	subPath := fmt.Sprintf("%s/%s/%s", "invoices", invoiceID, "retry_payment")
+	clientRequest := &ClientRequest{
+		Path:   subPath,
+		Result: &InvoiceResult{},
+	}
+
+	result, err := ir.client.PostWithoutBody(ctx, clientRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	if result != nil {
+		invoiceResult, ok := result.(*InvoiceResult)
+		if !ok {
+			return nil, &ErrorTypeAssert
+		}
+
+		return invoiceResult.Invoice, nil
+	}
+
+	return nil, nil
```

Comment:
Could be reversed to have:
Always nicer (but that's pretty much opinionated) to have early returns.

```suggestion
	if result == nil {
	        return nil, nil
	}
		
	invoiceResult, ok := result.(*InvoiceResult)
        if !ok {
		return nil, &ErrorTypeAssert
	}

	return invoiceResult.Invoice, nil
```

Reply:
hey @arthureichelberger thanks for the insight, to be consistent we'll keep the actual format since it's the same on each calls!
but we can do a refacto if it's more in the go best practices!

Reply:
It's just coding style, don't worry, no need for a refacto! 🙌